File: src/main/java/com/onlineshop/catalog/logic/Catalog.java
package com.onlineshop.catalog.logic;

import com.onlineshop.catalog.api.CatalogAggregate;
import com.onlineshop.catalog.api.events.*;
import com.onlineshop.catalog.entity.ItemEntity;
import org.jetbrains.annotations.Nullable;
import ru.quipy.domain.AggregateState;

import java.util.ArrayList;
import java.util.UUID;

public class Catalog implements AggregateState<UUID, CatalogAggregate> {
    private UUID id;

    @Nullable
    @Override
    public UUID getId() {
        return id;
    }

    private ArrayList<ItemEntity> Items = new ArrayList<>();



    public CreatedItemEvent addItemToCatalog(ItemEntity item) {
        if (Items.contains(item)) {
            throw new IllegalArgumentException("Item already exists");
        }
        return new CreatedItemEvent(item.getId(), item.getName(), item.getDescription(), item.getPrice(), item.getAmount());
    }

    public ItemRemovedFromTheCatalogEvent removeItemFromCatalog(ItemEntity item) {
        if (!Items.contains(item)) {
            throw new IllegalArgumentException("No such item");
        }
        Items.remove(item);
        return new ItemRemovedFromTheCatalogEvent(item.getId());
    }

    public ItemPriceChangedEvent changeItemPrice(ItemEntity item, int price) {
        if (!Items.contains(item)) {
            throw new IllegalArgumentException("No such item");
        }
        for (ItemEntity ouritem : Items) {
            if (ouritem.equals(item)) {
                ouritem.setPrice(price);
            }
        }
        return new ItemPriceChangedEvent(item.getId(), item.getPrice());
    }

    public ItemRefiledEvent refilItem(ItemEntity item, int amount) {
        if (!Items.contains(item)) {
            throw new IllegalArgumentException("No such item");
        }
        for (ItemEntity ourItem : Items) {
            if (ourItem.equals(item)) {
                ourItem.setAmount(amount);
            }
        }
        return new ItemRefiledEvent(item.getId(), item.getAmount());
    }
    //TODO переписать контролер
    //TODO проверьте правильность методов)
}

File: src/main/kotlin/com/onlineshop/catalog_service/logic/Catalog.kt
package com.onlineshop.catalog_service.logic

import com.onlineshop.catalog_service.api.*
import ru.quipy.domain.AggregateState
import java.lang.IllegalArgumentException
import java.util.*
import kotlin.collections.ArrayList

class Catalog : AggregateState<UUID, CatalogAggregate>{
    private lateinit var id: UUID
    var ItemList: MutableMap<UUID, Item> = mutableMapOf()

    override fun getId() = id

    fun createCatalog(id: UUID = UUID.randomUUID()) : CatalogCreatedEvent{
        return CatalogCreatedEvent(id)
    }

    fun addItemToCatalog (id: UUID, name : String, description: String, amount: Int, price: Int, category: String) : ItemAddedEvent{
        if(ItemList.containsKey(id)){
            throw IllegalArgumentException("Item with id $id is already exists")
        }
        //TODO хз, надо ли создавать итем и добавлять его в мапу
        return ItemAddedEvent(id, name, description, amount, price, category)
    }
    fun deleteIttemFromCatalog (id : UUID) : ItemRemovedEvent{
        if (!ItemList.containsKey(id)){
            throw IllegalArgumentException("No such item with id $id")
        }
        return ItemRemovedEvent(id)
    }
    fun changeItemPrice (id : UUID, price : Int) : ItemPriceChangedEvent{
        if (!ItemList.containsKey(id)){
            throw IllegalArgumentException("No such item with id $id")
        }
        return ItemPriceChangedEvent(id, price)
    }
    fun changeItemAmount(id : UUID, amount : Int) : ItemRefiledEvent{
        if (!ItemList.containsKey(id)){
            throw IllegalArgumentException("No such item with id $id")
        }
        return ItemRefiledEvent(id, amount)
    }
    //TODO не уверен надо делать изменение количества товара или только его пополнения, если что пофиксите
}

data class Item(
    val id : UUID,
    val name : String,
    val description : String,
    val price : Int,
    val amount : Int,
    val category : String,
)